app/src/main/java/nu/yona/app/ui/message/NotificationFragment.java

```diff
@@ -359,10 +359,14 @@ else if (mYonaMessages.getLinks().getNext() != null && loadMore)
 		{
 			urlForMessageFetch = mYonaMessages.getLinks().getNext();
 		}
-		else
+		else if (mYonaMessages.getPage().getTotalPages() > 1)
```

Comment:
What is the objective of this condition? And what is "first" here? The server returns ``next`` and ``previous`` links, but no ``first``, so where is that from?

Reply:
```"_links": {
		"first": {
			"href": "https://app.beta.yona.nu/users/936e36c9-35b7-4cfd-96be-a864ee1204aa/messages/?onlyUnreadMessages=true&page=0&size=3"
		},
		"self": {
			"href": "https://app.beta.yona.nu/users/936e36c9-35b7-4cfd-96be-a864ee1204aa/messages/?onlyUnreadMessages=true&page=0&size=3"
		},
		"next": {
			"href": "https://app.beta.yona.nu/users/936e36c9-35b7-4cfd-96be-a864ee1204aa/messages/?onlyUnreadMessages=true&page=1&size=3"
		},
		"last": {
			"href": "https://app.beta.yona.nu/users/936e36c9-35b7-4cfd-96be-a864ee1204aa/messages/?onlyUnreadMessages=true&page=1&size=3"
		},
		"curies": [{
			"href": "http://dev.yona.nu/rels/{rel}",
			"name": "yona",
			"templated": true
		}]
	},
	"page": {
		"size": 3,
		"totalElements": 5,
		"totalPages": 2,
		"number": 0
	}
```
I got the first and last links for the messages along with nex and previous, first will have value only if it has more that one page. 
When user clicks on notification, app will display the details. From there, if user clicks back button, app will go to notifications screen, here app was crashing as list previous state is not maintained at the moment. So to avoid the confusion I am displaying the first page again.

Reply:
Sorry, I've never seen that ``first`` link.
Anyway, the pagination approach sound wrong to me. You start from the first and go on till the end. If one goes backward, you go on till the beginning. Both are indicated by the availability of ``next`` and ``previous`` links.